gravitee-apim-gateway/gravitee-apim-gateway-policy/src/main/java/io/gravitee/gateway/policy/PolicyFactoryCreator.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.policy;
+
+import org.springframework.beans.factory.FactoryBean;
+
+/**
+ * Factory of PolicyFactory. A PolicyFactory is created for each new API deployed.
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface PolicyFactoryCreator extends FactoryBean<PolicyFactory> {
```

Comment:
Can you please try to avoid Spring things ?

Reply:
working on it. PlatformConfiguration injects a PolicyFactory so I was relying on it.
but does not work if I want to debug Platform level policies -> so refacto

Reply:
forced push to get rid of FactoryBean
for the moment Platform-level policies are not handled by debug mode. it will be handled here
https://github.com/gravitee-io/issues/issues/6983